functests-extended/1_performance_operator_upgrade/upgrade_operator.go

```diff
@@ -95,6 +117,18 @@ var _ = Describe("[rfe_id:28567][performance] Performance Addon Operator Upgrade
 		csv = getCSV(subscription.Status.CurrentCSV, namespaces.PerformanceOperator)
 		csvWaitForPhaseWithConditionReason(csv.Name, namespaces.PerformanceOperator, olmv1alpha1.CSVPhaseSucceeded, olmv1alpha1.CSVReasonInstallSuccessful)
 		Expect(csv.ObjectMeta.Annotations["containerImage"]).NotTo(Equal(fromImage))
+
+		// it is impossible to predict if it was some changes under generated by PAO KubeletConfig or Tuned
+		// during the PAO upgrade, so the best we can do here is to wait some time,
+		// to give other controllers time to notify changes
+		time.Sleep(2 * time.Minute)
+		mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+		var workerRTNodesNames []string
+		for _, workerRTNode := range workerRTNodes {
+			workerRTNodesNames = append(workerRTNodesNames, workerRTNode.Name)
+		}
+		Expect(tuned.WaitForAppliedCondition(workerRTNodesNames, corev1.ConditionTrue, 5*time.Minute))
```

Comment:
suggestion (ignore if this is handled): 
If we want to check that tuned daemonsets where reloaded before profile applied check 
we can keep their pids for example `/run/tuned/tuned.pid`  then check if it was changed (or check the pod generation)

Reply:
well its either that or the 2 minutes wait you added above, but because we also need to wait for kubeconfig and MCs to be updated then maybe we can keep the more general (not "precise") approach